docs/spec/spec.md

```diff
@@ -0,0 +1,950 @@
+    }
+}
+```
+
+## 4.4. Bidirectional Streaming RPC
+The RPC service definition of a bidirectional streaming call is as follows.
+```proto
+service RouteGuide {
+    rpc ListFeatures(Rectangle) returns (stream Feature) {}
+}
+```
+
+**RPC Using Direct Return**
+
+The Ballerina implementation of the bidirectional streaming RPC using a direct return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function RouteChat(stream<RouteNote, grpc:Error?> clientNotesStream) returns stream<RouteNote, grpc:Error?>|error {
+        RouteNote[] routeNotes = [];
+        check clientNotesStream.forEach(function(RouteNote note) {
+            ROUTE_NOTES.push(note);
+            foreach RouteNote n in ROUTE_NOTES {
+                if n.location == note.location {
+                    routeNotes.push(note);
+                }
+            }
+        });
+        return routeNotes.toStream();
+    }
+}
+```
+
+Note that, here using direct return will not address the exact use case. This example was added, only for completeness.
+
+**RPC Using a Caller**
+
+The Ballerina implementation of the bidirectional streaming RPC using a caller return is as follows.
+
+```ballerina
+service "RouteGuide" on new grpc:Listener(8980) {
+
+    remote function RouteChat(RouteGuideRouteNoteCaller caller, stream<RouteNote, grpc:Error?> clientNotesStream) returns error? {
+        check clientNotesStream.forEach(function(RouteNote note) {
+            future<error?> f1 = start sendRouteNotesFromLocation(caller, note.location);
+            lock {
+                ROUTE_NOTES.push(note);
+            }
+            error? waitErr = wait f1;
+        });
+    }
+}
+```
+
+# 5. gRPC Security
+
+## 5.1 Authentication and Authorization
+
+There are two ways to enable authentication and authorization in gRPC.
+1. Declarative approach
+2. Imperative approach
+
+### 5.1.1 Declarative Approach
+
+This is also known as the configuration-driven approach, which is used for simple use cases, where users have to provide a set of configurations and do not need to be worried more about how authentication and authorization works. The user does not have full control over the configuration-driven approach.
+
+The service and/or resource configurations are used to define the authentication and authorization configurations. Users can configure the configurations needed for different authentication schemes and configurations needed for authorizations of each authentication scheme. Also, the configurations can be provided at both the service and resource levels. The priority will be given from bottom to top. Then, the auth handler creation and request authentication/authorization is handled internally without user intervention. The requests that succeeded both authentication and/or authorization phases according to the configurations will be passed to the business logic layer.
```

Comment:
gRPC doesnt provide resource level configuration since there are no resource methods right?

Reply:
Fixed in here https://github.com/ballerina-platform/module-ballerina-grpc/pull/583/commits/b02c203ed54cffed7a481d25723eca7a422f1717